operator crashes after a few seconds

**What did you do?**
Install prometheus-operator 0.10.0

**What did you expect to see?**
A constantly running prometheus-operator pod

**What did you see instead? Under which circumstances?**
a failing operator pod


**Environment**

* Kubernetes version information:

	inClient Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.4", GitCommit:"d6f433224538d4f9ca2f7ae19b252e6fcb66a3ae", GitTreeState:"clean", BuildDate:"2017-05-19T18:44:27Z", GoVersion:"go1.7.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.2+coreos.0", GitCommit:"79fee581ce4a35b7791fdd92e0fc97e02ef1d5c0", GitTreeState:"clean", BuildDate:"2017-04-19T23:13:34Z", GoVersion:"go1.7.5", Compiler:"gc", Platform:"linux/amd64"}

* Kubernetes cluster kind: 

	bootkube

* Manifests:

```
kube-prometheus from this repo
```

* Prometheus Operator Logs:


```
s=2017-06-09T14:39:40Z caller=operator.go:220 component=prometheusoperator msg="connection established" cluster-version=v1.6.2+coreos.0
ts=2017-06-09T14:39:40Z caller=operator.go:1029 component=prometheusoperator msg="TPR created" tpr=service-monitor.monitoring.coreos.com
ts=2017-06-09T14:39:40Z caller=operator.go:536 component=alertmanageroperator msg="TPR created" tpr=alertmanager.monitoring.coreos.com
ts=2017-06-09T14:39:40Z caller=operator.go:1029 component=prometheusoperator msg="TPR created" tpr=prometheus.monitoring.coreos.com
ts=2017-06-09T14:39:43Z caller=operator.go:150 component=alertmanageroperator msg="TPR API endpoints ready"
ts=2017-06-09T14:39:47Z caller=operator.go:234 component=prometheusoperator msg="TPR API endpoints ready"
ts=2017-06-09T14:39:47Z caller=operator.go:619 component=prometheusoperator msg="sync prometheus" key=monitoring/k8s
ts=2017-06-09T14:39:47Z caller=operator.go:970 component=prometheusoperator msg="current config or current configmaps has changed"
ts=2017-06-09T14:39:47Z caller=operator.go:976 component=prometheusoperator msg="updating configuration"
ts=2017-06-09T14:39:47Z caller=operator.go:619 component=prometheusoperator msg="sync prometheus" key=rook/rook-prometheus
ts=2017-06-09T14:39:47Z caller=operator.go:967 component=prometheusoperator msg="updating config skipped, no configuration change"
ts=2017-06-09T14:39:47Z caller=operator.go:619 component=prometheusoperator msg="sync prometheus" key=monitoring/k8s
ts=2017-06-09T14:39:47Z caller=operator.go:970 component=prometheusoperator msg="current config or current configmaps has changed"
ts=2017-06-09T14:39:47Z caller=operator.go:976 component=prometheusoperator msg="updating configuration"
ts=2017-06-09T14:39:47Z caller=operator.go:584 component=prometheusoperator msg="update handler" old=6590300 cur=6590405
ts=2017-06-09T14:39:47Z caller=operator.go:321 component=alertmanageroperator msg="update handler" old=6590300 cur=6590405
ts=2017-06-09T14:39:47Z caller=operator.go:619 component=prometheusoperator msg="sync prometheus" key=monitoring/k8s
ts=2017-06-09T14:39:47Z caller=operator.go:967 component=prometheusoperator msg="updating config skipped, no configuration change"
ts=2017-06-09T14:39:47Z caller=operator.go:584 component=prometheusoperator msg="update handler" old=6590405 cur=6590406
ts=2017-06-09T14:39:47Z caller=operator.go:321 component=alertmanageroperator msg="update handler" old=6590405 cur=6590406
ts=2017-06-09T14:39:47Z caller=operator.go:619 component=prometheusoperator msg="sync prometheus" key=monitoring/k8s
ts=2017-06-09T14:39:47Z caller=operator.go:967 component=prometheusoperator msg="updating config skipped, no configuration change"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x11779e6]
 goroutine 344 [running]:
github.com/coreos/prometheus-operator/pkg/prometheus.(*prometheusCollector).collectPrometheus(0xc42032fac0, 0xc4202107e0, 0xc42033c010)
	/go/src/github.com/coreos/prometheus-operator/pkg/prometheus/collector.go:56 +0x86
github.com/coreos/prometheus-operator/pkg/prometheus.(*prometheusCollector).Collect(0xc42032fac0, 0xc4202107e0)
	/go/src/github.com/coreos/prometheus-operator/pkg/prometheus/collector.go:51 +0x87
github.com/coreos/prometheus-operator/vendor/github.com/prometheus/client_golang/prometheus.(*Registry).Gather.func2(0xc4202c21b0, 0xc4202107e0, 0x1c7e820, 0xc42032fac0)
	/go/src/github.com/coreos/prometheus-operator/vendor/github.com/prometheus/client_golang/prometheus/registry.go:433 +0x61
created by github.com/coreos/prometheus-operator/vendor/github.com/prometheus/client_golang/prometheus.(*Registry).Gather
	/go/src/github.com/coreos/prometheus-operator/vendor/github.com/prometheus/client_golang/prometheus/registry.go:434 +0x2ec
```

--- a/pkg/alertmanager/collector.go
+++ b/pkg/alertmanager/collector.go
@@ -53,5 +53,9 @@ func (c *alertmanagerCollector) Collect(ch chan<- prometheus.Metric) {
 }
 
 func (c *alertmanagerCollector) collectAlertmanager(ch chan<- prometheus.Metric, a *v1alpha1.Alertmanager) {
-	ch <- prometheus.MustNewConstMetric(descAlertmanagerSpecReplicas, prometheus.GaugeValue, float64(*a.Spec.Replicas), a.Namespace, a.Name)
+	replicas := float64(minReplicas)
+	if a.Spec.Replicas != nil {
+		replicas = float64(*a.Spec.Replicas)
+	}
+	ch <- prometheus.MustNewConstMetric(descAlertmanagerSpecReplicas, prometheus.GaugeValue, replicas, a.Namespace, a.Name)
 }

--- a/pkg/prometheus/collector.go
+++ b/pkg/prometheus/collector.go
@@ -53,5 +53,9 @@ func (c *prometheusCollector) Collect(ch chan<- prometheus.Metric) {
 }
 
 func (c *prometheusCollector) collectPrometheus(ch chan<- prometheus.Metric, p *v1alpha1.Prometheus) {
-	ch <- prometheus.MustNewConstMetric(descPrometheusSpecReplicas, prometheus.GaugeValue, float64(*p.Spec.Replicas), p.Namespace, p.Name)
+	replicas := float64(minReplicas)
+	if p.Spec.Replicas != nil {
+		replicas = float64(*p.Spec.Replicas)
+	}
+	ch <- prometheus.MustNewConstMetric(descPrometheusSpecReplicas, prometheus.GaugeValue, replicas, p.Namespace, p.Name)
 }

